pkg/route/api/helper.go

```diff
@@ -20,11 +22,33 @@ func RouteLessThan(route1, route2 *Route) bool {
 	if route1.CreationTimestamp.Before(route2.CreationTimestamp) {
 		return true
 	}
-	if route1.CreationTimestamp == route2.CreationTimestamp && route1.UID < route2.UID {
-		return true
+
+	if route1.CreationTimestamp == route2.CreationTimestamp {
+		if route1.UID < route2.UID {
+			return true
+		}
+		if route1.Namespace < route2.Namespace {
+			return true
+		}
+		return route1.Name < route2.Name
 	}
-	if route1.Namespace < route2.Namespace {
-		return true
+
+	return false
+}
+
+// GetSubdomainForHost returns the subdomain for the specified host.
+// This handles top-level domain names as well.
+// Example: acme.test and www.acme.test both return acme.test
+//          and www1.edge.acme.test returns edge.acme.test
+func GetSubdomainForHost(host string) string {
```

Comment:
If we don't do that then `acme.test` and `foo.test` would belong to the same domain for ownership purpose - we do need to handle TLDs separately.  And I did add similar test to the `host_admitter_test.go` and they pass. Will try your specific test as well but it might not be valid because of the `*.foo.com` host names - its a wee bit different now in terms of what the `host` field is set to.